# shippio OCR
## やること
- フォーマット判別
- OCR API比較
 - GCP（GCP, AWS, Azureの中で唯一OCRを提供)
 - Herokuで見つけたOCRestfulは精度が悪い
 - 自社API(tensorflowを用いる)
- 

## 使用する環境
- python 2.7
- tensorflow
- cv2(←openCV3です)

In [1]:
import tensorflow
import cv2


## GCP OCRにクエリを投げてみる


In [7]:
import requests
import json

def GCP(img):
    api_key = ""
    url = "https://vision.googleapis.com/v1/images:annotate?key=" + api_key

    d = {
      "requests":[
        {
          "image":{
            "source":{
              "imageUri":
                img
            }
          },
          "features":[
            {
              "type":"TEXT_DETECTION",
              "maxResults":1
            }
          ]
        }
      ]
    }
    r = requests.post(url, json=d)

    data=json.loads(r.text)

    #print json.dumps(data, indent=2)
    
    return data["responses"][0]["textAnnotations"][0]["description"]

In [8]:
from IPython.display import Image
from IPython.core.display import HTML 

img = "http://hajimek.com/export_serial.png"

result = GCP(img)

print result

Image(url=img)

3441025302186286



In [9]:
img = "http://hajimek.com/maker_serial.png"

result = GCP(img)

print result

Image(url=img)

NZ'I 2-6 Onl |||310 irdi2.6 5.8



In [10]:
img = "http://hajimek.com/monochrome.png"

result = GCP(img)

print result

Image(url=img)

NZT-2.60-indai:0:4:2658



# OCRest
## 下記を実行したところ
`curl -X POST --data-binary @/Users/hajimex/Documents/circlein/ocr/data/OCR_pic/XXXXX.png -H "secret:" -H "content-type:image/png" https://api.ocrestful.com/henam/res `

## 結果は
透かしなしの場合、confidenceは83%で、かつ3441025302186286
透かしありの場合、最後の”８”だけ精度が高かった

## 考察
精度は通常の画像は、GCPと同じ程度の精度が出ている。かつ、confidenceが出るのは良い。
ノイズに対してはGCPよりは弱い。